src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ModuleSelectionDialog.java

```diff
@@ -69,5 +76,16 @@ private void createUIComponents() {
     modulesComboBox.setRenderer(new IconListCellRenderer<>(viewModel.getPrompt(),
         Module::getName, PluginIcons.A_PLUS_MODULE));
     modulesComboBox.selectedItemBindable.bindToSource(viewModel.selectedModule);
+    fileChooser = new FieldPanel(getText("ui.moduleSelectionDialog.path"), null, event -> browse(), null);
+    viewModel.selectedModuleFile.addValueObserver(this,
+        (self, file) -> ((FieldPanel) fileChooser).setText(Optional.ofNullable(file).map(
+            VirtualFile::getPath).orElse("")));
+    ((FieldPanel) fileChooser).setEditable(false);
+  }
+
+  private void browse() {
+    var descriptor = FileChooserDescriptorFactory.createSingleLocalFileDescriptor();
+    FileChooser.chooseFile(descriptor, viewModel.getProject(), basePanel, viewModel.selectedModuleFile.get(),
+        viewModel.selectedModuleFile::set);
   }
```

Comment:
Remark: Again ideally, this logic would be handled in a domain-independent subclass of FieldPanel that would have a a Bindable to be bound to the view model.